frame/examples/offchain-worker-ping-pong/src/lib.rs

```diff
@@ -0,0 +1,681 @@
+			// after `next_unsigned_at` will ever get to the transaction pool
+			// and will end up in the block. We can still have multiple
+			// transactions compete for the same "spot", and the one with higher
+			// priority will replace other one in the pool.
+			.and_provides(next_unsigned_at)
+			// The transaction is only valid for next 5 blocks. After that it's
+			// going to be revalidated by the pool.
+			.longevity(5)
+			// It's fine to propagate that transaction to other peers, which
+			// means it can be created even by nodes that don't produce blocks.
+			// Note that sometimes it's better to keep it for yourself (if you
+			// are the block producer), since for instance in some schemes
+			// others may copy your solution and claim a reward.
+			.propagate(true)
+			.build()
+	}
+
+	/// A helper function to send a signed pong transaction from the OCW.
+	fn ocw_pong_signed() -> Result<(), &'static str> {
+		let signer = Signer::<T, T::AuthorityId>::all_accounts();
+		if !signer.can_sign() {
+			return Err(
+				"No local accounts available. Consider adding one via `author_insertKey` RPC.",
+			)
+		}
+
+		let pings = <Pings<T>>::get();
+		for p in pings {
+			let Ping(nonce) = p;
+
+			// Using `send_signed_transaction` associated type we create and
+			// submit a transaction representing the call, we've just created.
+			// Submit signed will return a vector of results for all accounts
+			// that were found in the local keystore with expected `KEY_TYPE`.
+			let results = signer.send_signed_transaction(|_account| {
+				// nonce is wrapped into a call to `pong_signed` public function
+				// of this pallet. This means that the transaction, when
+				// executed, will simply call that function passing `nonce` as
+				// an argument.
+				Call::pong_signed { nonce }
+			});
+
+			for (acc, res) in &results {
+				match res {
+					Ok(()) => log::info!("[{:?}] Submitted pong with nonce {}", acc.id, nonce),
+					Err(e) => log::error!("[{:?}] Failed to submit transaction: {:?}", acc.id, e),
+				}
+			}
+		}
+
+		Ok(())
+	}
+
+	/// A helper function to sign payload and send an unsigned pong transaction
+	fn ocw_pong_unsigned_for_any_account(
+		block_number: BlockNumberFor<T>,
+	) -> Result<(), &'static str> {
+		let pings = <Pings<T>>::get();
+		for p in pings {
+			let Ping(nonce) = p;
+
+			// -- Sign using any account
+			let (_, result) = Signer::<T, T::AuthorityId>::any_account()
+				.send_unsigned_transaction(
+					|account| PongPayload { nonce, block_number, public: account.public.clone() },
+					|payload, signature| Call::pong_unsigned_with_signed_payload {
+						pong_payload: payload,
+						signature,
+					},
+				)
+				.ok_or("No local accounts accounts available.")?;
+			result.map_err(|()| "Unable to submit transaction")?;
+		}
+
+		Ok(())
+	}
+
+	/// A helper function to sign payload and send an unsigned pong transaction
+	fn ocw_pong_unsigned_for_all_accounts(
```

Comment:
Why do we need this function twice, just for `any`/`all` accounts?

Reply:
I'm not really sure what the practical difference is either.

this is simply following the same pattern as the original OCW example:
https://github.com/paritytech/substrate/blob/28b1f79abedbad2ec97b49277b36e1f797ca9e5d/frame/examples/offchain-worker/src/lib.rs#L511

https://github.com/paritytech/substrate/blob/28b1f79abedbad2ec97b49277b36e1f797ca9e5d/frame/examples/offchain-worker/src/lib.rs#L541